tests/test_experimental.py

```diff
@@ -630,6 +631,290 @@ def multiply(self, X, K):
+    distances_2d = np.sqrt((xx - x_obs) ** 2 + (yy - y_obs) ** 2)
+    distances = distances_2d.flatten()
+    rho = np.exp(-0.5 * (distances / localization_radius) ** 2).reshape(-1, 1)
+
+    # --- Run Assimilations ---
+    # Initialize smoothers with the current run's seeded RNG
+    esmda_distance = DistanceESMDA(
+        covariance=C_D, observations=true_observations, alpha=alpha_i, seed=rng
+    )
+    X_posterior = esmda_distance.assimilate(X=X_prior, Y=Y, rho=rho)
+
+    esmda_global = ESMDA(
+        covariance=C_D, observations=true_observations, alpha=alpha_i, seed=rng
+    )
+    X_posterior_global = esmda_global.assimilate(X=X_prior, Y=Y)
+
+    # --- Assertions ---
+    # Find parameters far from the observation where localization weight is near zero
+    zero_weight_indices = np.where(rho < 1e-6)[0]
+    assert len(zero_weight_indices) > 0, (
+        "No distant points found for testing localization."
+    )
+
+    # --- Assertions for Localized Smoother ---
+    prior_mean = np.mean(X_prior, axis=1)
+    posterior_mean = np.mean(X_posterior, axis=1)
+    prior_variance = np.var(X_prior, axis=1)
+    posterior_variance = np.var(X_posterior, axis=1)
+
+    # Assert distant parameters are not updated
+    assert np.allclose(
+        X_posterior[zero_weight_indices, :], X_prior[zero_weight_indices, :], rtol=1e-2
+    )
+    # Assert parameter at observation IS updated
+    assert not np.allclose(X_posterior[flat_obs_index, :], X_prior[flat_obs_index, :])
+
+    # Assert update is maximal at the observation point
+    update_magnitude = np.abs(posterior_mean - prior_mean)
+    assert np.argmax(update_magnitude) == flat_obs_index
+
+    # Assert variance is reduced at observation, but preserved far away
+    assert posterior_variance[flat_obs_index] < prior_variance[flat_obs_index]
+    assert np.allclose(
+        posterior_variance[zero_weight_indices], prior_variance[zero_weight_indices]
+    )
+
+    # --- Comparison Assertions for Global Smoother ---
+    posterior_mean_global = np.mean(X_posterior_global, axis=1)
+    posterior_variance_global = np.var(X_posterior_global, axis=1)
+
+    # Assert distant parameters ARE updated due to spurious correlations
+    assert not np.allclose(
+        X_posterior_global[zero_weight_indices, :],
+        X_prior[zero_weight_indices, :],
+    )
+
+    # Assert AVERAGE variance is reduced everywhere (the sign of ensemble collapse)
+    assert np.mean(posterior_variance_global[zero_weight_indices]) < np.mean(
+        prior_variance[zero_weight_indices]
+    )
+
+    # --- Overall Quality Assertion (MSE) ---
+    # Assert that the localized result is a more plausible estimate
+    mse_localized = np.mean((posterior_mean - true_parameters) ** 2)
+    mse_non_localized = np.mean((posterior_mean_global - true_parameters) ** 2)
+    assert mse_localized < mse_non_localized
+
+
+@pytest.mark.parametrize("seed", list(range(9)))
+def test_distance_based_localization_on_3D_case(seed):
```

Comment:
There is this sufffix everywhere `_with_single_observation` witch either should be dropped and provided in all tests?